manticore/platforms/evm.py

```diff
@@ -1645,6 +1645,10 @@ def CALLDATACOPY(self, mem_offset, data_offset, size):
         if issymbolic(size):
             raise ConcretizeStack(3, policy='ALL')
 
+        if not issymbolic(data_offset):
+            data_offset = min(data_offset, len(self.data))
+            size = min(len(self.data) - data_offset, size)
+
         for i in range(size):
             c = Operators.ITEBV(8, data_offset + i < len(self.data), Operators.ORD(self.data[data_offset + i]), 0)
```

Comment:
This looks redundant with the above checks, but in the symbolic case, we still want this bounds check.